pydantic/_internal/_model_construction.py

```diff
@@ -35,7 +35,7 @@
 object_setattr = object.__setattr__
 
 
-def init_private_attributes(self_: Any, **_kwargs: Any) -> None:
+def init_private_attributes(self_: Any, context: Any) -> None:
```

Comment:
Note the `__pydantic_post_init__` function is called with `context=None` in various places, but it is now also called with positional arguments from Rust, so we _have_ to name this argument `context` and accept it positionally.

If we want to change the way `init_private_attributes` is called from python (e.g., in `pydantic.main.BaseModel.__init__`, with `context=None`), I'm okay either way. Doesn't need to happen in this PR, but it can.